### DISCUSSION TITLE
Can we use filter in subcribe or subcribe to collection based on some field

### DISCUSSION THREAD

**Discussion by MuhammadKhizar7** [pocketbase/pocketbase#3539] [Q&A]
We have chat messages collection if we subscribe to this, like
```
Subscribe to changes in any messages record
pb.collection('messages').subscribe('*', function (e) {
    console.log(e.action);
    console.log(e.record);
});

```
It will send event on create, delete, and update this collection Let's say we have collection messages in relation to chatrooms and we have multiple chatrooms that have messages above the code that send messages to every subscriber even when no chat is happening in that chatroom. Can we somehow restrict this subscribe collection to only that chatroom based on chat_id present in message?

or we can subscribe to chatroom collection with relation to messages. When message collection changes, it will send messages to that chatroom or maybe some other solution

chat -> id, users
messages -> id, content, chat_id (chat), author(user)

**Comment by ganigeorgiev** [comment]
`filter`, `expand`, and other request parameters support for the realtime subscriptions are planned for the next release (_there are no ETAs yet_).

At the moment you can perform the filter on the client-side inside the callback.

Note that when subscribing to the wildcard topic (`*`) the List API rule of the collection will be executed to check whether the subscribed client is allowed to retrieve the change.

**Reply by MuhammadKhizar7** [reply]
Thanks for quick reply
